src/Orleans/Providers/IOrleansProvider.cs

```diff
@@ -160,4 +161,23 @@ public static TimeSpan GetTimeSpanProperty(this IProviderConfiguration config, s
             return config.Properties.TryGetValue(key, out s) ? TimeSpan.Parse(s) : settingDefault;
         }
     }
+
+    /// <summary>
+    /// Exception thrown whenever a provider has failed to be initialized.
+    /// </summary>
+    [Serializable]
+    public class ProviderInitializationException : OrleansException
```

Comment:
We added it here because it's an addition to the public surface of the provider model.  On initialization failure, providers are expected to throw this exception.